### ISSUE TITLE
Memory leak in controller manager

### ISSUE THREAD

**Issue by julia-stripe** [kubernetes/kubernetes#52657]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

Apologies for this not being more specific from "in the controller manager" -- I've attached a heap dump which hopefully will clarify but I haven't figured out yet how to track this down further to a specific controller. Right now we're restarting the controller manager every hour to mitigate this issue.

**What happened**:

The controller manager leaked 32GB of memory before being OOM killed. Here's a screenshot from our monitoring tool:

<img width="652" alt="screen shot 2017-09-18 at 9 00 18 am" src="https://user-images.githubusercontent.com/23065472/30551658-ff5e29b2-9c4f-11e7-8551-40e304bd7d69.png">

We have quite a lot of pod churn in our cluster because we run cron jobs, which I believe is related to this. There are not very many *active* pods in our cluster at any given time -- at most ~300. 

Here's a heap profile from pprof:

* PDF image of the heap profile: [profile004.pdf](https://github.com/kubernetes/kubernetes/files/1311320/profile004.pdf)
* raw data: 
 [pprof.com_github_kubernetes_hyperkube.static.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz](https://github.com/kubernetes/kubernetes/files/1311228/pprof.com_github_kubernetes_hyperkube.static.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz)

I'm not super experienced with reading pprof files but this looks to me like pods are being leaked somewhere. There are about 1.2GB of pods, which is way more than the 300 active pods in the cluster should account for. 

Here's a summary of the pprof file above:

```
(pprof) top
Showing nodes accounting for 1914.61MB, 96.89% of 1976.01MB total
Dropped 328 nodes (cum <= 9.88MB)
Showing top 10 nodes out of 49
      flat  flat%   sum%        cum   cum%
  778.09MB 39.38% 39.38%   778.09MB 39.38%  runtime.rawstringtmp
  597.51MB 30.24% 69.62%  1211.27MB 61.30%  k8s.io/kubernetes/pkg/api/v1.(*PodSpec).Unmarshal
  183.11MB  9.27% 78.88%   494.25MB 25.01%  k8s.io/kubernetes/pkg/api/v1.(*Container).Unmarshal
  160.57MB  8.13% 87.01%   170.78MB  8.64%  runtime.mapassign
   88.07MB  4.46% 91.46%    88.07MB  4.46%  reflect.unsafe_New
      34MB  1.72% 93.19%       56MB  2.83%  k8s.io/kubernetes/pkg/api/v1.(*VolumeSource).Unmarshal
   27.01MB  1.37% 94.55%    66.01MB  3.34%  k8s.io/kubernetes/pkg/api/v1.(*PodStatus).Unmarshal
   19.05MB  0.96% 95.52%    19.05MB  0.96%  runtime.makemap
      17MB  0.86% 96.38%   541.01MB 27.38%  k8s.io/apimachinery/pkg/apis/meta/v1.(*ObjectMeta).Unmarshal
   10.21MB  0.52% 96.89%    10.21MB  0.52%  runtime.hashGrow
```

**Environment**:

- Kubernetes version: 1.7.3
- Cloud provider: AWS
- OS: Ubuntu 16.04
- Kernel: 4.4

**Comment by julia-stripe** [comment]
/sig api-machinery
(not sure if that's right)

**Comment by dims** [comment]
@julia-stripe - Do you have `successfulJobsHistoryLimit ` and `failedJobsHistoryLimit ` set? See tip here : https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#jobs-history-limits

```
By default, there are no limits, and all successful and failed jobs are kept. However, jobs can pile up quickly when running a cron job, and setting these fields is recommended
```

**Comment by dims** [comment]
@soltysh thoughts?

**Comment by julia-stripe** [comment]
@dims 

We do have `successfulJobsHistoryLimit` and `failedJobsHistoryLimit` set.

If the problem were due to the actual number of jobs in the cluster, I'd expect the memory usage to immediately go back up once the controller manager is restarted. The fact that restarting it seems to fix the problem makes me think it's a leak. Does that reasoning make sense to you?

**Comment by cheftako** [comment]
@julia-stripe Quick sanity check, you didn't turn on ENABLE_CACHE_MUTATION_DETECTOR did you? (The only way I have been able to reproduce what you describe so far was by turning on the mutation detector)

**Comment by julia-stripe** [comment]
Nope! Happy to do anything that would help to try to figure this out though

**Comment by cheftako** [comment]
What do you get when you run "kubectl describe jobs/<job name>"

**Comment by julia-stripe** [comment]
Here are the overall counts of the number of pods/jobs/cronjobs in the cluster. These numbers are pretty typical.

```
$ kubectl get pods --all-namespaces -a | wc -l
273
$ kubectl get jobs --all-namespaces -a | wc -l
2028
$ kubectl get cronjobs --all-namespaces -a | wc -l
193
```

**Comment by julia-stripe** [comment]
Here's another screenshot of our memory usage graphs showing this is still an issue. We have a local cronjob that kill -9s the controller manager at 43 minutes after the hour (to keep the memory usage under control), so memory usage rises steadily until :43 and then drops when the controller manager is killed, and then it starts rising again. 

<img width="600" alt="screen shot 2017-10-02 at 7 25 35 am" src="https://user-images.githubusercontent.com/23065472/31082164-fd9ea54a-a742-11e7-8968-af4f42888fad.png">

**Comment by cheftako** [comment]
cc @jennybuckley

**Comment by cheftako** [comment]
We have reproduced the problem locally and we believe we have tracked down where the memory is accumulating. Now we are working on actually fixing it.

**Comment by jennybuckley** [comment]
@julia-stripe 

I've been working on this bug for a while, and trying to understand what might be going on. When I run a cron job, I can't seem to get the cache to fill indefinitely. When the job completes a certain number of times (successfulJobsHistoryLimit), the old versions seem to be cleaned up and the cache size stableizes. I wonder if maybe the jobs created by your set up are never completing, and thus never getting cleaned up.

If you examine the jobs running on your cluster after letting it run for a while, are any of them not completed and very old?

**Comment by dekhtyarev** [comment]
Hello!

We are faced with a similar problem.

**Environment**:

* Kubernetes version: 1.6.7
* Cloud provider: our VMs
* OS: Ubuntu 16.04
* Kernel: 4.4

We have 2 controllers in the cluster.
Schedule of RAM consumption for the last 24 hours:

![image](https://user-images.githubusercontent.com/17539237/31707504-bc89ea02-b416-11e7-9d50-5a16978d4d49.png)

The active controller stably eats our memory.
We have already limited their 512 MB of memory and now the kubelet solves the problem with the leak. But it would be great to get rid of it completely.

---

We have cron jobs, 10+ per day.
Sometimes jobs are cleaned, sometimes not.

Number of pods:
```
$ kubectl get pods --all-namespaces | wc -l
629
```

Number of completed jobs:
```
$ kubectl get pods --all-namespaces --show-all | grep "Completed" | wc -l
78
```

The oldest job: 22 days

---

What other information can we provide to help solve this problem?

**Comment by joewilliams** [comment]
We may also be seeing this issue on:
* Kube 1.6.7
* Debian 8 (jessie)
* linux 4.8.0-0.bpo.2-amd64

We have no cronjobs.

**Comment by jamiehannaford** [comment]
@cheftako In your [local testing](https://github.com/kubernetes/kubernetes/issues/52657#issuecomment-330716414), where in the CM was the leak occurring? Did you identify a fix?

**Comment by jennybuckley** [comment]
@jamiehannaford 
We think the issue isn't so much that the CM is leaking memory, but that the job controller manager is actually designed to rely on an ever increasing cache in order to figure out how many pods have completed. ([here](https://github.com/kubernetes/kubernetes/blob/f34d3af809169007e7f474a26751ee675c0664ac/pkg/controller/job/job_controller.go#L624)). This decision was made because of the assumption that a the controller manager should be able to be shut down, be restarted, and the job's status should be able to be reconstructed from the state of the cluster.

The information about which pods have succeeded and which have failed should probably be persisted in a more sustainable way though, so large jobs don't cause the memory to fill up indefinitely.
@kow3ns is there anyone on your team who would be interested in looking more into this? It seems like there's not a clear single owner of the Job Controller code.

**Comment by soltysh** [comment]
There's an open issue to improve job controller not to rely on pod being around, see https://github.com/kubernetes/kubernetes/issues/28486 

> @kow3ns is there anyone on your team who would be interested in looking more into this? It seems like there's not a clear single owner of the Job Controller code.

As one of the main authors of jobs and cronjobs I'd be more than happy to help you with that, but I won't be able to implement it myself, due to time contraints. Maybe for 1.11 :/

**Comment by kvaps** [comment]
Hi, same problem occurs last week, each 1-2 days.

![image](https://user-images.githubusercontent.com/7556217/37024944-9f9d6668-212a-11e8-8b60-aa24878f1e8f.png)

**Environment:**

*    Kubernetes version:  1.9.3
*    Cloud provider: our environment (Kubernetes runs inside LXC-Conteiner)
*    OS: Ubuntu 16.04
*    Kernel: 4.13.13-5

**Comment by naveens** [comment]
Issue still happens in our environment with only ~200 pods, no jobs/cronjobs.

**Environment:**
    Kubernetes version: 1.7.8
    Cloud provider: On-Prem
    OS: Container Linux by CoreOS (1409.6.0)
    Kernel: 4.11.9

```(pprof) top
Showing nodes accounting for 10.94GB, 98.60% of 11.09GB total
Dropped 327 nodes (cum <= 0.06GB)
Showing top 10 nodes out of 32
      flat  flat%   sum%        cum   cum%
    3.51GB 31.64% 31.64%     3.55GB 31.97%  runtime.mapassign
    2.66GB 23.94% 55.58%     2.66GB 23.94%  runtime.rawstringtmp
    1.93GB 17.41% 73.00%     7.86GB 70.82%  k8s.io/kubernetes/pkg/api/v1.(*PodSpec).Unmarshal
    0.87GB  7.87% 80.86%     0.87GB  7.87%  reflect.unsafe_New
    0.78GB  7.08% 87.94%     5.42GB 48.85%  k8s.io/kubernetes/pkg/api/v1.(*Container).Unmarshal
    0.34GB  3.06% 91.00%     0.34GB  3.06%  runtime.makemap
    0.32GB  2.84% 93.84%     0.63GB  5.64%  k8s.io/kubernetes/pkg/api/v1.(*PodStatus).Unmarshal
    0.26GB  2.38% 96.21%     0.32GB  2.85%  k8s.io/kubernetes/pkg/api/v1.(*Handler).Unmarshal
    0.17GB  1.56% 97.77%     0.24GB  2.18%  k8s.io/kubernetes/pkg/api/v1.(*VolumeSource).Unmarshal
    0.09GB  0.83% 98.60%     1.61GB 14.53%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.(*ObjectMeta).Unmarshal```

**Comment by joewilliams** [comment]
As a workaround for this we put a 16Gi memory limit in the `kube-controller-manager` config and that has worked well for over a month now. Hope this helps.

**Comment by tamalsaha** [comment]
Hi,
We are noticing a similar issue in our CRD controller https://github.com/appscode/voyager/issues/1004 . Was anyone able to find a root cause for this issue?

Thanks.

**Comment by alok87** [comment]
@justinsb How can we put restrictions using kops for controller manager. We are also facing the leak
$ kubectl top po -n kube-system | grep controller-man
kube-controller-manager-ip-A.ap-southeast-1.compute.internal   0m           15Mi
kube-controller-manager-ip-B-ap-southeast-1.compute.internal    0m           15Mi
kube-controller-manager-ip-C-ap-southeast-1.compute.internal   108m         3478Mi

**Comment by soltysh** [comment]
@alok87 can you try to nail down which exact controller is causing this problem, by trying to disable each resource (if possible). Usually if the resource is disabled, the controller responsible for that is not started. This could allow us to figure out which exact controller should be blamed for.

**Comment by liggitt** [comment]
If you are rapidly creating and deleting pods containing tolerations, a memory leak was just found in the node lifecycle taint controller. See https://github.com/kubernetes/kubernetes/pull/65339 for details

**Comment by liggitt** [comment]
backport PRs for anyone interested in tracking the fix back to a release branch:

Automated cherry pick of #65339: Remove item from taint manager workqueue on completion 
- 1.11: #65362
- 1.10: #65363
- 1.9: #65364
- 1.8: #65365

**Comment by joshphp** [comment]
I am also having issues with `kube-controller-manager` using lots of memory (3.8GBish).

Kubernetes version: 1.9.7
Cloud provider: AWS
OS: Debian9

**Comment by jjo** [comment]
`v1.9.8` -> `v1.9.9` seems to have fixed `kube-controller-manager` RSS:
![k dev_kube-controller-manager_rss 20180718](https://user-images.githubusercontent.com/88727/42905271-f02d1136-8aad-11e8-8c8c-763c914e587d.png)

Our use case creates in the order of 1k~ish per hour `Pods` from app-driven scheduled Kubernetes `Jobs`, with tolerations indeed (to land these at specific nodes).

**Comment by liggitt** [comment]
closing as resolved by https://github.com/kubernetes/kubernetes/pull/65339

/close

**Comment by zjzhangkui** [comment]
@liggitt  is it normal the controller-manager continuous consumed 4.7g memory on 8G node?  

I see a lot of [anon]  by "pmap -x 1" command.

1: kube-controller-manager --address=127.0.0.1 --leader-elect=true --use-service-account-credentials=true --service-account-private-key-file=/etc/kubernetes/pki/sa.key --cluster-signing-cert-file=/etc/kubernetes/pki/ca.crt --cluster-signing-key-file=/etc/kub
Address		  Kbytes     PSS   Dirty    Swap  Mode  Mapping
0000000000400000   37320    5132       0       0  r-xp  /usr/local/bin/kube-controller-manager
0000000002872000   39756    5688       0       0  r--p  /usr/local/bin/kube-controller-manager
0000000004f45000     984     344     260      28  rw-p  /usr/local/bin/kube-controller-manager
000000000503b000     188     120     120       0  rw-p    [ anon ]
000000c000000000    4620    4620    4620       0  rw-p    [ anon ]
000000c416fbe000  153864  153864  153864       0  rw-p  [stack:7]
000000c420600000    2048    1416    1412     360  rw-p  [stack:13]
000000c420800000    4096    2996    2996     400  rw-p    [ anon ]
000000c420c00000    6144    3768    3760    1744  rw-p    [ anon ]
000000c421200000    2048     928     928     804  rw-p  [stack:10]
000000c421400000    4096    2020    2020    1588  rw-p    [ anon ]
000000c421800000    6144    5120    5120      88  rw-p  [stack:15]
000000c421e00000   22528   17336   17296    4568  rw-p  [stack:14]
000000c423400000    1024     832     832     192  rw-p    [ anon ]
000000c423500000    1024     836     836     188  rw-p    [ anon ]
000000c423600000    3072    2620    2616     452  rw-p    [ anon ]
000000c423900000    1024     896     896     128  rw-p    [ anon ]
000000c423a00000   23552   20384   20328    3136  rw-p    [ anon ]
000000c425100000    1024     892     888     132  rw-p    [ anon ]
000000c425200000    1024     924     924     100  rw-p    [ anon ]
000000c425300000    1024     888     888     136  rw-p    [ anon ]
000000c425400000    1024     832     832     192  rw-p    [ anon ]
000000c425500000    1024     644     644     380  rw-p    [ anon ]
000000c425600000    3072    2608    2608     464  rw-p    [ anon ]
000000c425900000    1024     904     904     120  rw-p    [ anon ]
000000c425a00000    8192    7164    7164    1028  rw-p    [ anon ]
000000c426200000    4096    3520    3516     576  rw-p    [ anon ]
000000c426600000   12288   11144   11144    1144  rw-p    [ anon ]
000000c427200000    6144    5092    5092    1052  rw-p    [ anon ]
000000c427800000    2048    1748    1744     300  rw-p    [ anon ]
000000c427a00000    1024     928     928      96  rw-p    [ anon ]
000000c427b00000    1024     864     864     160  rw-p    [ anon ]
000000c427c00000    3072    2820    2816     252  rw-p    [ anon ]
000000c427f00000    1024     912     912     112  rw-p    [ anon ]
000000c428000000    9216    7708    7704    1508  rw-p    [ anon ]
000000c428900000    1024     872     872     152  rw-p    [ anon ]
000000c428a00000    5120    4272    4272     848  rw-p    [ anon ]
000000c428f00000    2048    1688    1688     360  rw-p    [ anon ]
000000c429100000    1024     908     904     116  rw-p    [ anon ]
000000c429200000    5120    4360    4356     760  rw-p    [ anon ]
000000c429700000    1024     904     904     120  rw-p    [ anon ]
000000c429800000    6144    5348    5348     796  rw-p    [ anon ]
000000c429e00000    2048    1724    1720     324  rw-p    [ anon ]
000000c42a000000    1024     824     820     200  rw-p    [ anon ]
000000c42a100000    1024     832     832     192  rw-p    [ anon ]
000000c42a200000   13312   11232   11220    2080  rw-p    [ anon ]
000000c42af00000    1024     884     884     140  rw-p    [ anon ]
000000c42b000000   16384   14264   14256    2120  rw-p    [ anon ]
000000c42c000000    2048    1628    1628     420  rw-p    [ anon ]
000000c42c200000    1024     848     848     176  rw-p    [ anon ]
000000c42c300000    1024     884     884     140  rw-p    [ anon ]
000000c42c400000   16384   13924   13912    2460  rw-p    [ anon ]
000000c42d400000    6144    5032    5032    1112  rw-p    [ anon ]
000000c42da00000    6144    5104    5100    1040  rw-p    [ anon ]
000000c42e000000    7168    5952    5952    1216  rw-p    [ anon ]

**Comment by zjzhangkui** [comment]
@redbaron Most is cost by Unmarshal, there is not much pod in k8s cluster,  less than 50 pods.
(pprof) top50 -cum
Showing nodes accounting for 4.02GB, 96.78% of 4.15GB total
Dropped 237 nodes (cum <= 0.02GB)
      flat  flat%   sum%        cum   cum%
         0     0%     0%     4.05GB 97.65%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch.(*StreamWatcher).receive
         0     0%     0%     4.05GB 97.65%  k8s.io/kubernetes/vendor/k8s.io/client-go/rest/watch.(*Decoder).Decode
         0     0%     0%     4.05GB 97.63%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/protobuf.(*Serializer).Decode
         0     0%     0%     4.05GB 97.63%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*DirectDecoder).Decode
         0     0%     0%     4.05GB 97.63%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.DirectDecoder.Decode
         0     0%     0%     4.05GB 97.61%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.Decode
         0     0%     0%     4.05GB 97.57%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/protobuf.unmarshalToObject
         0     0%     0%     3.50GB 84.44%  k8s.io/kubernetes/vendor/github.com/gogo/protobuf/proto.Unmarshal
         0     0%     0%     3.50GB 84.44%  k8s.io/kubernetes/vendor/github.com/gogo/protobuf/proto.UnmarshalMerge
         0     0%     0%     3.50GB 84.38%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Pod).Unmarshal
    1.17GB 28.17% 28.17%     3.02GB 72.85%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodSpec).Unmarshal
    0.47GB 11.41% 39.58%     1.60GB 38.45%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Container).Unmarshal
    0.85GB 20.54% 60.12%     0.86GB 20.79%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*ResourceRequirements).Unmarshal
         0     0% 60.12%     0.54GB 13.13%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.(*Scheme).New
         0     0% 60.12%     0.54GB 13.13%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.UseOrCreateObject
         0     0% 60.12%     0.54GB 13.13%  reflect.New
    0.54GB 13.13% 73.25%     0.54GB 13.13%  reflect.unsafe_New
    0.33GB  7.86% 81.11%     0.33GB  7.86%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.(*ObjectMeta).Unmarshal
    0.03GB  0.79% 81.90%     0.18GB  4.27%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Volume).Unmarshal
    0.17GB  4.13% 86.03%     0.17GB  4.13%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*EnvVar).Unmarshal
    0.08GB  1.99% 88.01%     0.15GB  3.68%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodStatus).Unmarshal
    0.09GB  2.22% 90.24%     0.14GB  3.48%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*VolumeSource).Unmarshal
    0.09GB  2.12% 92.36%     0.09GB  2.12%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*VolumeMount).Unmarshal
    0.08GB  1.85% 94.20%     0.08GB  1.85%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Toleration).Unmarshal
    0.05GB  1.25% 95.45%     0.05GB  1.25%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodCondition).Unmarshal
         0     0% 95.45%     0.05GB  1.09%  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1
         0     0% 95.45%     0.03GB  0.76%  k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*processorListener).(k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.run)-fm
         0     0% 95.45%     0.03GB  0.76%  k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*processorListener).run
    0.03GB  0.75% 96.20%     0.03GB  0.75%  k8s.io/kubernetes/pkg/controller/node/scheduler.(*NoExecuteTaintManager).PodUpdated
    0.02GB  0.58% 96.78%     0.02GB  0.58%  k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*SecretVolumeSource).Unmarshal

**Comment by xigang** [comment]
> @redbaron Most is cost by Unmarshal, there is 45 pods created in cluster，calculate by command kubectl get po --all-namespaces -a| wc -l We had once frequently create pods when testing cicd jobs 2 months ago, but the memory will not going down until today. Tolerations is added by default as joshphp said. Maybe the same issue as fixed in #65339
> 
> (pprof) top50 -cum Showing nodes accounting for 4.02GB, 96.78% of 4.15GB total Dropped 237 nodes (cum <= 0.02GB) flat flat% sum% cum cum% 0 0% 0% 4.05GB 97.65% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/watch.(*StreamWatcher).receive 0 0% 0% 4.05GB 97.65% k8s.io/kubernetes/vendor/k8s.io/client-go/rest/watch.(*Decoder).Decode 0 0% 0% 4.05GB 97.63% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/protobuf.(*Serializer).Decode 0 0% 0% 4.05GB 97.63% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.(*DirectDecoder).Decode 0 0% 0% 4.05GB 97.63% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.DirectDecoder.Decode 0 0% 0% 4.05GB 97.61% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.Decode 0 0% 0% 4.05GB 97.57% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime/serializer/protobuf.unmarshalToObject 0 0% 0% 3.50GB 84.44% k8s.io/kubernetes/vendor/github.com/gogo/protobuf/proto.Unmarshal 0 0% 0% 3.50GB 84.44% k8s.io/kubernetes/vendor/github.com/gogo/protobuf/proto.UnmarshalMerge 0 0% 0% 3.50GB 84.38% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Pod).Unmarshal 1.17GB 28.17% 28.17% 3.02GB 72.85% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodSpec).Unmarshal 0.47GB 11.41% 39.58% 1.60GB 38.45% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Container).Unmarshal 0.85GB 20.54% 60.12% 0.86GB 20.79% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*ResourceRequirements).Unmarshal 0 0% 60.12% 0.54GB 13.13% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.(*Scheme).New 0 0% 60.12% 0.54GB 13.13% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime.UseOrCreateObject 0 0% 60.12% 0.54GB 13.13% reflect.New 0.54GB 13.13% 73.25% 0.54GB 13.13% reflect.unsafe_New 0.33GB 7.86% 81.11% 0.33GB 7.86% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.(*ObjectMeta).Unmarshal 0.03GB 0.79% 81.90% 0.18GB 4.27% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Volume).Unmarshal 0.17GB 4.13% 86.03% 0.17GB 4.13% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*EnvVar).Unmarshal 0.08GB 1.99% 88.01% 0.15GB 3.68% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodStatus).Unmarshal 0.09GB 2.22% 90.24% 0.14GB 3.48% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*VolumeSource).Unmarshal 0.09GB 2.12% 92.36% 0.09GB 2.12% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*VolumeMount).Unmarshal 0.08GB 1.85% 94.20% 0.08GB 1.85% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*Toleration).Unmarshal 0.05GB 1.25% 95.45% 0.05GB 1.25% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*PodCondition).Unmarshal 0 0% 95.45% 0.05GB 1.09% k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1 0 0% 95.45% 0.03GB 0.76% k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*processorListener).(k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.run)-fm 0 0% 95.45% 0.03GB 0.76% k8s.io/kubernetes/vendor/k8s.io/client-go/tools/cache.(*processorListener).run 0.03GB 0.75% 96.20% 0.03GB 0.75% k8s.io/kubernetes/pkg/controller/node/scheduler.(*NoExecuteTaintManager).PodUpdated 0.02GB 0.58% 96.78% 0.02GB 0.58% k8s.io/kubernetes/vendor/k8s.io/api/core/v1.(*SecretVolumeSource).Unmarshal

@zjzhangkui Was this issue resolved later?
